### PR TITLE
[FIX] Added JSON Formatting check for custom translations field 

### LINKED ISSUE #27846: Custom Translation allows invalid json and break UI
### Description:

We wanted to make some changes using custom translation, we forgot to include the `<br/>` tag at the beginning, so the front end had broken

### Steps to reproduce:

1. Settings
2. General
3. Translations
4. Custom translations
5. Set the field with: `<code>{ "en": { "Channels": "Rooms" }, "pt": { "Channels": "Salas" } }</code>`  ignoring the` <br/>
`
the next output is how is setted in the setting document in mongo:
```
 db.rocketchat_settings.find({_id : "Custom_Translations"}).pretty()
{
        "_id" : "Custom_Translations",
        "createdAt" : ISODate("2022-12-05T20:37:53.542Z"),
        "value" : "<code>{ \"en\": { \"Welcome\": \"Bienvenidos\" } }</code>",
        "packageValue" : "",
        "valueSource" : "packageValue",
        "secret" : false,
        "enterprise" : false,
        "i18nDescription" : "Custom_Translations_Description",
        "autocomplete" : true,
        "sorter" : 25000,
        "ts" : ISODate("2022-12-05T20:37:53.542Z"),
        "type" : "code",
        "group" : "General",
        "section" : "Translations",
        "public" : true,
        "i18nLabel" : "Custom_Translations",
        "hidden" : false,
        "blocked" : false,
        "requiredOnWizard" : false,
        "env" : false,
        "_updatedAt" : ISODate("2023-01-25T13:26:36.722Z")
}
```
notice the lack of the `<br/>` tag in the value field

### Expected behavior:

The front end still alive

### Actual behavior:
The front end breaks

image

### Server Setup Information:

Implementación
Versión 5.4.1
ID de implementación BY5JmCq5iXeezaP5e
Versión de motor de aplicaciones 1.36.0
Versión de nodo v14.19.3
Migración de base de datos
281 (25 de enero de 2023 10:28)
MongoDB 5.0.8 / wiredTiger (oplog Habilitada)
Detalles de confirmación HEAD: (f7908e8e8)
Rama: HEAD
PID 751170

### Client Setup Information

Happens only in web and desktop app

### Workarround
Change the setting by mongosh

Open your OS terminal

open monsgo console

`> mongo`

Run this query to show the values in the DB

`rs01:PRIMARY> db.rocketchat_settings.find({_id : \"Custom_Translations\"}).pretty()`

run this query to clean the custom field

`rs01:PRIMARY> db.rocketchat_settings.update({"_id":"Custom_Translations"},{$set:{"value":""}})`

### PR DESCRIPTION
**Pull request by Pradumn27**
## Proposed changes (including videos or screenshots)

Currently the UI is breaking when an invalid JSON is sent in custom translations. So, I've added the check for valid JSON on the frontend to avoid the UI breaking due to that. Along with that I've also added the valid JSON Formatting check on the saveSettings call at the backend thus ensuring end to end checks for this field.
I've also changed the help text field here to avoid the confusion caused due to that.

https://user-images.githubusercontent.com/47187878/215126463-20801be3-aa39-4867-8ba1-ab2eb0505bc6.mov

## Issue(s)

Closes #27846

## Steps to test or reproduce

1. Settings
2. General
3. Translations
4. Custom Translations
5. Add an invalid JSON to the field

### CODE REVIEW

**Comment by Pradumn27** [comment]
@dudanogueira can you review and merge the PR, Thank you.

File: apps/meteor/app/lib/server/methods/saveSettings.js [hunk 1]
```diff
@@ -41,6 +52,9 @@ Meteor.methods({
 					case 'multiSelect':
 						check(value, Array);
 						break;
+					case 'code':
+						parseToJSON(value);
```
Comment (matheusbsilva137) [inline apps/meteor/app/lib/server/methods/saveSettings.js:56]: Not all `code` settings are JSON (eg some of them are CSS or HTML). We should check settings' `code` property to assure they should be a valid JSON.

**Review by matheusbsilva137 — COMMENTED** [review]
Heeey @Pradumn27 ! Thanks for suggesting these changes
Please check #28600

**Comment by dougfabris** [comment]
Closing this one in favor of: #28600
Thanks for the contribution @Pradumn27

**Pull request closed without merging**